backend/api/routes/events.js

```diff
@@ -55,6 +56,51 @@ router.route("/").post(authenticateMiddleware, async (req, res) => {
     res
       .status(201)
       .json({ success: true, message: "event was added to database!" });
+
+    //background processing of notifications after response is sent
+    if (privacy_level === "private" && invited_users.length > 0) {
+      invited_users.forEach((userId) => {
+        createNotification(supabaseClient, {
+          userId: userId,
+          type: "private_event_invite",
+          title: "You are invited to a Private Event!",
+          message: `@${req.user.user_metadata.name} invited you to "${title}"`,
+          priority: "immediate",
+        });
+      });
+    } else if (privacy_level === "network") {
+      const { data: connections } = await supabaseClient
+        .from("connections")
+        .select("connected_streamer_id")
+        .eq("user_id", creator_id);
+
+      connections?.forEach((connection) => {
+        createNotification(supabaseClient, {
+          userId: connection.connected_streamer_id,
+          type: "network_event_announcement",
+          title: "New Event from Your Network",
+          message: `@${req.user.user_metadata.name} created "${title}" for people in their network`,
+          priority: "general",
+        });
+      });
+    } else if (privacy_level === "public") {
+      const { data: allUsers } = await supabaseClient
+        .from("profiles")
+        .select("id");
+
+      allUsers?.forEach((user) => {
+        // Don't notify the creator
+        if (user.id !== creator_id) {
+          createNotification(supabaseClient, {
+            userId: user.id,
+            type: "public_event_announcement",
+            title: "New Public Event Available",
+            message: `@${req.user.user_metadata.name} created "${title}" for all check it out!`,
+            priority: "general",
+          });
+        }
+      });
```

Comment:
Let's add a todo here to revisit this notification, you mention that you should only notify users on the location or with similar interests, right?